geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-parquet/src/main/scala/org/locationtech/geomesa/fs/storage/parquet/ParquetFileSystemStorage.scala

```diff
@@ -81,6 +83,32 @@ object ParquetFileSystemStorage {
       observer.write(f)
     }
     override def flush(): Unit = observer.flush()
-    override def close(): Unit = CloseQuietly(Seq(writer, observer)).foreach(e => throw e)
+    override def close(): Unit = {
+      println(s"Closing file ${file}")
+      CloseQuietly(Seq(writer, observer)).foreach(e => throw e)
+      validateParquetFile(file)
+    }
+  }
+
+  def validateParquetFile(file: Path): Unit = {
+    if (!fileValidationEnabled.get.toBoolean) {
```

Comment:
i'd suggest moving this outside the function - that means we only have to convert it to a boolean once, and also makes the function a little more logical i think, i.e. you'd expect the function to validate a file but it actually won't if the sys prop isn't set

Reply:
yeah, we're not actually testing that the method gets invoked appropriately in the close() method. but i don't think we can test that case unless we can figure out a good way to corrupt the file after the parquet writer is closed but before the file is validated. since the logic we're not testing is fairly simple, i'd prefer having the method be intuitive, rather than testing the boolean if check.